File: Taller de Haskell/Problema 5/problema5.hs
tipoImpuesto :: Int -> Float -> String
tipoImpuesto edad ingresos
    | ingresos > 5000000 = "Debe pagar impuesto de renta"
    | edad > 18 && ingresos > 1000000 = "Debe pagar impuesto salarial"
    | otherwise = "No debe pagar impuestos"
    
main :: IO ()
main = do
    putStrLn "Ingrese su edad: "
    entradaEdad <- getLine
    putStrLn "Ingrese sus ingresos mensuales (en COP): "
    entradaIngresos <- getLine

    let edad = read entradaEdad :: Int
    let ingresos = read entradaIngresos :: Float

    putStrLn (tipoImpuesto edad ingresos)

File: Taller de Haskell/Problema 5/problema5.py
# Le pedimos la edad y los ingresos al usuario
edad = int(input("Ingrese su edad: "))
ingresos = float(input("Ingrese sus ingresos mensuales (en COP): "))

# Se verifica qué tipo de impuesto debe pagar
if ingresos > 5000000:
    print("Debe pagar impuesto de renta")
elif edad > 18 and ingresos > 1000000:
    print("Debe pagar impuesto salarial")
else:
    print("No debe pagar impuestos")
